walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,67 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
+
+  /** Returns time ranges that satisfy the request and take into account, the events */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long duration = request.getDuration();
+    // meeting duration is greater than a day means no viable ranges
+    if (duration > TimeRange.WHOLE_DAY.duration()) return Arrays.asList();
```

Comment:
nit: always prefer curly bracket syntax.  See the story of Apple's 2014 #gotofail bug: https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/